packages/firebase_dynamic_links/android/src/main/java/io/flutter/plugins/firebasedynamiclinks/FirebaseDynamicLinksPlugin.java

```diff
@@ -27,15 +31,30 @@
 import java.util.Map;
 
 /** FirebaseDynamicLinksPlugin */
-public class FirebaseDynamicLinksPlugin implements MethodCallHandler, NewIntentListener {
-  private final Registrar registrar;
-  private final MethodChannel channel;
+public class FirebaseDynamicLinksPlugin
+    implements FlutterPlugin, ActivityAware, MethodCallHandler, NewIntentListener {
+  private Registrar registrar;
+  private MethodChannel channel;
+  private ActivityPluginBinding activityBinding;
 
   private FirebaseDynamicLinksPlugin(Registrar registrar, MethodChannel channel) {
```

Comment:
You added annotations right? Tag Registrar nonnull. Even better, since you only pull the activity out of the registrar, just change the instance variable to activity so future maintainers don't accidentally do registrar.messenger() and expand the encapsulation in the future and trip on a null half of the time.

Reply:
I removed the annotations in this PR because the ecosystem team decided that we won't include androidx in any plugin that doesn't require it. It's so that a Flutter app that doesn't include androidx can use this plugin. 

However, this is a good point about someone accidently using the registrar. What about adding the `@Deprecated` annotation with an explanation about registrar remaining until we decide to only support v2 embedding flutter apps.

Reply:
This does remind me that I also need to add documentation for the plugin now. Updated PR

Reply:
Sorry, this tab was open and I missed the new comments. I agree with Maurice's summary about annotation, commented [here](https://github.com/FirebaseExtended/flutterfire/pull/1372#discussion_r345966532).

For `@Deprecated`, I vaguely remember floating the idea by @amirh offline and he was against it. The reason being that for now the static registrar registerWith method _is_ fully supported, and we don't want to flag it as deprecated before we're actually really deprecating it. I think those uses will show up as compile time warnings too, so probably better not to bring them in.

Reply:
We don't want to `@Deprecate` the actual Registrar API in the engine like @mklim said. 
If you meant `@Deprecate` the instance variable, that sounds ok. I don't have a strong opinion. I was just making a general comment (regardless of what the class is) to minimize access. If your plugin instance wants an activity, just keep an activity and not its provider.

Reply:
I agree. I remember that we originally kept an instance of `Registrar` and not `Activity` because the method `Registrar.getActivity()` would sometimes return different activities in add-to-app scenarios. 

Since we're now migrating to v2 embedding, I don't this this would be the case anymore. I updated the plugin to hold a reference to `Activity`.